x-pack/plugins/secops/server/lib/types.ts

```diff
@@ -35,3 +38,86 @@ export interface Configuration {
 export interface Context {
   req: FrameworkRequest;
 }
+
+export type ESQuery = ESRangeQuery | ESQueryStringQuery | ESMatchQuery | JsonObject;
+export type FilterQuery = JsonObject;
+
+export interface ESRangeQuery {
+  range: {
+    [name: string]: {
+      gte: number;
+      lte: number;
+      format: string;
+    };
+  };
+}
+
+export interface ESMatchQuery {
+  match: {
+    [name: string]: {
+      query: string;
+      operator: string;
+      zero_terms_query: string;
+    };
+  };
+}
+
+export interface ESQueryStringQuery {
+  query_string: {
+    query: string;
+    analyze_wildcard: boolean;
+  };
+}
+
+export interface SearchResponse<T> {
+  took: number;
+  timed_out: boolean;
+  _scroll_id?: string;
+  _shards: ShardsResponse;
+  hits: {
+    total: number;
+    max_score: number;
+    hits: Array<{
+      _index: string;
+      _type: string;
+      _id: string;
+      _score: number;
+      _source: T;
+      _version?: number;
+      _explanation?: Explanation;
+      fields?: string[];
+      // tslint:disable-next-line:no-any
+      highlight?: any;
+      // tslint:disable-next-line:no-any
+      inner_hits?: any;
+      matched_queries?: string[];
+      sort?: string[];
+    }>;
+  };
+  // tslint:disable-next-line:no-any
+  aggregations?: any;
+}
+
+export interface ShardsResponse {
+  total: number;
+  successful: number;
+  failed: number;
```

Comment:
Do you think we need a card in the backlog for bubbling up errors like this to the user?

Reply:
I agree and we have an issue to bubble up errors, it is in the backlog.